src/ui/submission.rs

```diff
@@ -16,15 +16,16 @@ pub fn print_submission_info(
     print_info_header(ui, &submission_info)?;
     let mut compiler_report_printed = print_compiler_report(ui, &submission_info)?;
     print_status(ui, &submission_info)?;
+    let mut spin = '|';
     while submission_info.pending {
+        spin = spinner(ui, spin)?;
```

Comment:
If taking `&mut char` instead,
```suggestion
        spinner(ui, &mut spin)?;
```